### PR TITLE
rpc: add getter trait methods to `ReceiptResponse`

### LINKED ISSUE #1245: [Feature] Add getter trait methods to `ReceiptResponse`
### Component

rpc

### Describe the feature you would like

Add getters for all fields of `alloy_rpc_types_eth::TransactionReceipt` as trait methods of `ReceiptResponse`. This is needed to replace concrete types with network specific receipt type in reth.

https://github.com/alloy-rs/alloy/blob/37b0c294a2baa5e2ead36259f9b5c7d9a000ad7c/crates/rpc-types-eth/src/transaction/receipt.rs#L9-L65

https://github.com/alloy-rs/alloy/blob/37b0c294a2baa5e2ead36259f9b5c7d9a000ad7c/crates/network-primitives/src/traits.rs#L6-L26

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tcoratger**
Should close #1245

### CODE REVIEW

File: crates/network-primitives/src/traits.rs [hunk 1]
```diff
@@ -23,6 +27,36 @@ pub trait ReceiptResponse {
...
+    fn blob_gas_price(&self) -> Option<u128>;
+
+    /// Address of the sender.
+    fn from(&self) -> Address;
+
+    /// Address of the receiver.
+    fn to(&self) -> Option<Address>;
+
+    /// Post-transaction state root.
+    fn state_root(&self) -> Option<B256>;
```
Comment (mattsse) [inline crates/network-primitives/src/traits.rs:56]: we can remove this, this isn't filled anymore afaik

File: crates/network-primitives/src/traits.rs [hunk 2]
```diff
@@ -23,6 +27,36 @@ pub trait ReceiptResponse {
...
+
+    /// Authorization list.
+    fn authorization_list(&self) -> Option<Vec<SignedAuthorization>>;
```
Comment (mattsse) [inline crates/network-primitives/src/traits.rs:59]: this should be &[] otherwise this requires a clone

**Review by mattsse — CHANGES_REQUESTED** [review]
nits

all of these should be fine 

we also need to impl this for 

https://github.com/alloy-rs/alloy/blob/122c4b187af3c621bdd10f829d44538300c0dae0/crates/network/src/any/mod.rs#L72-L72

I believe

File: crates/network-primitives/src/traits.rs (hunk 1 above)
Comment (tcoratger) [reply crates/network-primitives/src/traits.rs]: fixed

File: crates/network-primitives/src/traits.rs (hunk 2 above)
Comment (tcoratger) [reply crates/network-primitives/src/traits.rs]: fixed

**Review by mattsse — APPROVED** [review]
all of these are good

**Pull request merged**